alf/algorithms/agent.py

```diff
@@ -272,74 +265,45 @@ def calc_training_reward(self, external_reward, info: AgentInfo):
         Returns:
             reward used for training.
         """
-        # record shaped extrinsic rewards actually used for training
-        self.add_reward_summary("reward/extrinsic", external_reward)
-
-        reward = external_reward
-        if self._extrinsic_reward_coef != 1.0:
-            reward *= self._extrinsic_reward_coef
-
-        if self._icm is not None:
-            self.add_reward_summary("reward/icm", info.icm.reward)
-            reward += self._intrinsic_reward_coef * info.icm.reward
-
-        if (self._goal_generator is not None
-                and 'reward' in info.goal_generator._fields):
-            self.add_reward_summary("reward/goal_generator",
-                                    info.goal_generator.reward)
-            reward += info.goal_generator.reward
-
-        if id(reward) != id(external_reward):
-            self.add_reward_summary("reward/overall", reward)
-
-        return reward
+        rewards = [
+            external_reward, info.icm.reward if self._icm else None,
+            info.goal_generator.reward if
+            (self._goal_generator and
+             ('reward' in info.goal_generator._fields)) else None
+        ]
+        weights = [
+            self._extrinsic_reward_coef, self._intrinsic_reward_coef, 1.0
+        ]
+        names = ["extrinsic", "icm", "goal_generator"]
+
```

Comment:
It's better to make code related to one algorithm adjacent to each other to minimize the effort of adding new algorithm.
```
rewards = [("extrinsic", external_reward, self._extrinsic_reward_conef)]
if self._icm:
   rewards.append(("icm", info.icm.reward, self._intrinsic_reward_coef))
if self._goal_generator and 'reward' in info.goal_generator._fields:
    rewards.append(("goal_generator", info.goal_generator.reward, 1.))
```